### ISSUE TITLE
[Builder.FullNodeFeature] System.OverflowException: Arithmetic operation resulted in an overflow

### ISSUE THREAD

**Issue by D3m0nKingx** [stratisproject/StratisBitcoinFullNode#1005]
```bash
fail: Stratis.Bitcoin.Builder.FullNodeFeatureExecutor[0]
      An error occurred starting the application: System.AggregateException: One or more errors occurred. (Arithmetic operation resulted in an overflow.) ---> System.OverflowException: Arithmetic operation resulted in an overflow
         at DBreeze.Storage.FSR.Table_Read(Boolean useCache, Int64 offset, Int32 count)
         at DBreeze.LianaTrie.LTrieWriteCache.ReadKey(Boolean useCache, Byte[] pointer)
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<IterateBackward>d__11.MoveNext()
         at DBreeze.Transactions.Transaction.<SelectBackward>d__80`2.MoveNext()
         at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
         at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
         at Stratis.Bitcoin.Features.Consensus.CoinViews.DBreezeCoinView.GetRewindIndex(Transaction transaction) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs:line 284
         at Stratis.Bitcoin.Features.Consensus.CoinViews.DBreezeCoinView.<Rewind>b__17_0() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs:line 301
         at System.Threading.Tasks.Task`1.InnerInvoke()
         at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
         at Stratis.Bitcoin.Features.Consensus.CoinViews.CachedCoinView.<Rewind>d__34.MoveNext() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs:line 418
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
         at Stratis.Bitcoin.Features.Consensus.ConsensusLoop.<StartAsync>d__36.MoveNext() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs:line 234
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Stratis.Bitcoin.Features.Consensus.ConsensusFeature.Initialize() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/ConsensusFeature.cs:line 113
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.<>c.<Initialize>b__3_1(IFullNodeFeature service) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 57
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Execute(Action`1 callback, Boolean reverseOrder) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 102
         --- End of inner exception stack trace ---
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Execute(Action`1 callback, Boolean reverseOrder) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 120
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Initialize() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 57
      ---> (Inner Exception #0) System.OverflowException: Arithmetic operation resulted in an overflow.
         at DBreeze.Storage.FSR.Table_Read(Boolean useCache, Int64 offset, Int32 count)
         at DBreeze.LianaTrie.LTrieWriteCache.ReadKey(Boolean useCache, Byte[] pointer)
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<ItBwd>d__10.MoveNext()
         at DBreeze.LianaTrie.Iterations.Backward.<IterateBackward>d__11.MoveNext()
         at DBreeze.Transactions.Transaction.<SelectBackward>d__80`2.MoveNext()
         at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
         at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
         at Stratis.Bitcoin.Features.Consensus.CoinViews.DBreezeCoinView.GetRewindIndex(Transaction transaction) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs:line 284
         at Stratis.Bitcoin.Features.Consensus.CoinViews.DBreezeCoinView.<Rewind>b__17_0() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs:line 301
         at System.Threading.Tasks.Task`1.InnerInvoke()
         at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
         at Stratis.Bitcoin.Features.Consensus.CoinViews.CachedCoinView.<Rewind>d__34.MoveNext() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs:line 418
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
         at Stratis.Bitcoin.Features.Consensus.ConsensusLoop.<StartAsync>d__36.MoveNext() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs:line 234
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Stratis.Bitcoin.Features.Consensus.ConsensusFeature.Initialize() in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin.Features.Consensus/ConsensusFeature.cs:line 113
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.<>c.<Initialize>b__3_1(IFullNodeFeature service) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 57
         at Stratis.Bitcoin.Builder.FullNodeFeatureExecutor.Execute(Action`1 callback, Boolean reverseOrder) in /home/ader/coins/StratisBitcoinFullNode/src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs:line 102<---
      
There was a problem initializing the node. Details: 'One or more errors occurred. (Arithmetic operation resulted in an overflow.)'
Application is shutting down...
```
Running StratisD on Testnet
LogFile: [debug.txt](https://github.com/stratisproject/StratisBitcoinFullNode/files/1594127/debug.txt)

**Comment by Aprogiena** [comment]
does it now occur to you every time you want to run this node?
if so, can you provide full data directory content?

**Comment by Aprogiena** [comment]
at first glance, this looks very much like https://github.com/hhblaze/DBreeze/issues/1, which should have been fixed a long time ago, so possibly another similar issue? 

it looks like it is not in our codebase, but if we want to ask hhblaze for help, we must be able to reproduce it

**Comment by D3m0nKingx** [comment]
Closing this issue and #1006 per discussion with @Aprogiena that we've determined this to be most likely caused by a corruption in database files while transferring the datadir from a node running on a remote server using scp (secure copy). Steps that caused the problem:

Quoted from slack by @Aprogiena:
1. machine A: node runs, node closes normally (this I call run K-1)
2. transfer of data directory from machine A to machine B (using scp command from remote -> local machine)
3. machine B: run node, node crashes on this very bug (this I call run K)
4. further attempts to run node with this data dir on machine B fails
5. a new attempt to transfer data directory from A to B does not reproduce the problem after run
